### PR TITLE
feat(deferAds): toggle defer ads

### PR DESCRIPTION
**Pull request by sivteck**
fixes https://github.com/quintype/madrid/issues/3601

### CODE REVIEW

File: js/utils.js [hunk 1]
```diff
@@ -18,10 +18,20 @@ function doloadGPTScript(resolve, reject, limitedAds) {
...
 }
 
-export function loadGPTScript(limitedAds = false) {
-  return new Promise((resolve, reject) => {
+export function loadGPTScript(limitedAds = false, deferAds = true) {
+  if (deferAds) {
+    return new Promise((resolve, reject) => {
+      const timerId = setTimeout(() => {
+        doloadGPTScript(resolve, reject, limitedAds, timerId);
+      }, 2500);
```
Comment (EnKrypt) [inline js/utils.js]: Can the delay amount be made as a variable? Using it as a config value would be good to have.

File: lib/utils.js [hunk 2]
```diff
@@ -26,10 +27,24 @@ function doloadGPTScript(resolve, reject, limitedAds) {
...
   document.getElementsByTagName('head')[0].appendChild(scriptTag);
+
+  if (timerId !== null) {
+    clearTimeout(timerId);
+  }
 }
 
 function loadGPTScript() {
   var limitedAds = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : false;
+  var deferAds = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : true;
```
Comment (EnKrypt) [inline lib/utils.js]:
```suggestion
  var limitedAds = arguments.length > 0 && arguments[0] ? arguments[0] : false;
  var deferAds = arguments.length > 1 && arguments[1] ? arguments[1] : true;
```

Having the field directly in the condition should be enough if we are checking if the value exists.

If we want to strictly check for `undefined`, we need to make use of `typeof`. The way it is written in the library is an anti pattern.

File: lib/idletimeout-shim.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+          return Math.max(0, 50.0 - (Date.now() - startTime));
+        }
+      });
+    }, 1);
+  };
+
+  return window.requestIdleCallback || fallback;
+};
+
+exports.getRequestIdleCallback = getRequestIdleCallback;
```
Comment (EnKrypt) [inline lib/idletimeout-shim.js:24]: How are we using this?

**Review by EnKrypt — CHANGES_REQUESTED** [review]

File: lib/idletimeout-shim.js (hunk 3 above)
Comment (sivteck) [reply lib/idletimeout-shim.js:24]: lib/* files are build output from js/* files

**Review by EnKrypt — APPROVED** [review]

**Review by Jeevan-Kishore — APPROVED** [review]

File: js/utils.js [hunk 4]
```diff
@@ -3,7 +3,9 @@ const GPT_SRC = {
   limitedAds: 'pagead2.googlesyndication.com',
 };
 
-function doloadGPTScript(resolve, reject, limitedAds) {
+const delayGPTScriptLoadTimeBy = 2500;
```
Comment (Harshith46) [inline js/utils.js]: why is this value hardcoded, can't this sent from the app level as config

**Review by Harshith46 — CHANGES_REQUESTED** [review]

**Review by EnKrypt — APPROVED** [review]

**Review by Harshith46 — APPROVED** [review]

**Review by nandap4790 — APPROVED** [review]

**Pull request merged**